gradio/reload.py

```diff
@@ -3,22 +3,28 @@
 Contains the functions that run when `gradio` is called from the command line. Specifically, allows
 
 $ gradio app.py, to run app.py in reload mode where any changes in the app.py file or Gradio library reloads the demo.
-$ gradio app.py my_demo, to use variable names other than "demo"
+$ gradio app.py my_demo.app, to use variable names other than "demo"
```

Comment:
Can we make this change backwards compatible by keeping the demo name without the ".app" suffix. Instead, we can add the suffix internally in `_setup_config()`

Reply:
I don't see a nice way to do it. Uvicorn is not triggering any Exepction I could hatch on to try one or the other.
Since this is a cli primary used for development, I'm not sure such breaking change is too much of an issue.

Reply:
I don't quite understand sorry. What do you mean "The problem with adding the .app is for factories. It won't work in these cases."

Reply:
Ok yeah it doesn't seem like Uvicorn throws an exception until the web app is actually running. So for now, I've added a warning if someone passes in a string that does not include "." for the app name, letting them know that the syntax has deprecated